### ISSUE TITLE
Glew 2.1.0 Does not build with VS2019

### ISSUE THREAD

**Issue by DieSlower** [nigels-com/glew#238]
I've tried to build glew 2.1.0 with VS2019 but I'm getting all kinds of errors. 
I'm using cmake 3.14.3

When I press configure, cmakes tells me it is done, but when I tell it to generate the cmake project it errors out. 

The Errors I'm getting are:
Configuring done
CMake Error at CMakeLists.txt:181 (add_executable):
  Cannot find source file:

    D:/Development/ExtDev/Build/glew-master/src/glewinfo.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx

CMake Error at CMakeLists.txt:107 (add_library):
  Cannot find source file:

    D:/Development/ExtDev/Build/glew-master/include/GL/wglew.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx

CMake Error at CMakeLists.txt:191 (add_executable):
  Cannot find source file:

    D:/Development/ExtDev/Build/glew-master/build/visualinfo.rc

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx

CMake Error at CMakeLists.txt:181 (add_executable):
  No SOURCES given to target: glewinfo

CMake Error at CMakeLists.txt:107 (add_library):
  No SOURCES given to target: glew

CMake Error at CMakeLists.txt:111 (add_library):
  No SOURCES given to target: glew_s

CMake Error at CMakeLists.txt:191 (add_executable):
  No SOURCES given to target: visualinfo

**Comment by nigels-com** [comment]
Try: https://github.com/Perlmint/glew-cmake

**Comment by DieSlower** [comment]
Hi Nigel,
That repo does not have a 2.1.0 tag. 
Is there a stable release of the 2.1.0 that works with VS2019?

**Comment by nigels-com** [comment]
Does GLEW 2.1.0 release zip build with VS2019?
I would be surprised if it didn't.
https://sourceforge.net/projects/glew/files/glew/2.1.0/

**Comment by DieSlower** [comment]
@nigels-com sorry for taking some time to reply, I just got back to this...
When I build from https://sourceforge.net/projects/glew/files/glew/2.1.0/ with vs2019 i get the following error:

   Creating library D:/Development/ExtDev/Build/glew-2.1.0/build/cmake/build/lib/Release/glew32.lib and object D:/Development/ExtDev/Build/glew-2.1.0/build/cmake/build/lib/Release/glew32.exp
glew.obj : error LNK2019: unresolved external symbol memset referenced in function glewContextInit
D:\Development\ExtDev\Build\glew-2.1.0\build\cmake\build\bin\Release\glew32.dll : fatal error LNK1120: 1 unresolved externals

**Comment by DieSlower** [comment]
Just FYI, this is the one that worked for me...
https://github.com/Perlmint/glew-cmake/tree/glew-cmake-2.1.0
I don't know what is different about it...it does seem to have a multi context and single context dll?

**Comment by nigels-com** [comment]
There is a new snapshot available.  Builds for me, for you?
https://sourceforge.net/projects/glew/files/glew/snapshots/glew-20190928.tgz/download

**Comment by DieSlower** [comment]
@nigels-com Yes, that snapshot builds great with VS2019! 
I noticed that the snapshot is version 2.2.0 ...when will that be officially released?

**Comment by matham** [comment]
I am trying to build the snapshot using 2019 from Github actions and it's erroring out with `error LNK2019: unresolved external symbol memset referenced in function glewContextInit`.

I'm using the [snapshot](http://jaist.dl.sourceforge.net/project/glew/glew/snapshots/glew-20190928.tgz). And the code is basically:

```
call "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\Tools\\VsDevCmd.bat"
devenv /upgrade .\\build\\vc12\\glew.sln
msbuild .\\build\\vc12\\glew.sln /property:Configuration=Release /property:Platform=x64
```

The error is:

```
glew.obj : error LNK2019: unresolved external symbol memset referenced in function glewContextInit [D:\a\kivy-sdk-packager\kivy-sdk-packager\kivy_build\glew\glew-2.2.0\build\vc12\glew_shared.vcxproj]

..\..\bin\Release\x64\glew32.dll : fatal error LNK1120: 1 unresolved externals [D:\a\kivy-sdk-packager\kivy-sdk-packager\kivy_build\glew\glew-2.2.0\build\vc12\glew_shared.vcxproj]

Done Building Project "D:\a\kivy-sdk-packager\kivy-sdk-packager\kivy_build\glew\glew-2.2.0\build\vc12\glew_shared.vcxproj" (default targets) -- FAILED.
```

The full build details is [here](https://github.com/kivy/kivy-sdk-packager/runs/365875345).

I'm hesitant to use https://github.com/Perlmint/glew-cmake because I'm not sure exactly how it relates to this repo?

**Comment by nigels-com** [comment]
@matham Could you try the vc15 projects, rather than vc12?

**Comment by matham** [comment]
>Could you try the vc15 projects, rather than vc12?

Yes, that indeed seems to work. I thought I looked into the folder and didn't see vc15 there before 🤦
